openml/datasets/functions.py

```diff
@@ -423,6 +481,24 @@ def create_dataset(name, description, creator, contributor,
     class:`openml.OpenMLDataset`
         Dataset description."""
 
+    if attributes == 'auto' or isinstance(attributes, dict):
+        if not hasattr(data, "columns"):
+            raise ValueError("Automatically inferring the attributes required "
+                             "a pandas DataFrame. A {!r} was given instead."
+                             .format(data))
+        # infer the type of data for each column of the DataFrame
+        attributes_ = attributes_arff_from_df(data)
+        if isinstance(attributes, dict):
+            # override the attributes which was specified by the user
+            for attr_idx in range(len(attributes_)):
+                attr_name = attributes_[attr_idx][0]
+                if attr_name in attributes.keys():
+                    attributes_[attr_idx] = (attr_name, attributes[attr_name])
+    else:
+        attributes_ = attributes
+
+    data = data.values if hasattr(data, "columns") else data
```

Comment:
This seems to imply that internally we store everything as arff. Wouldn't it be much cooler to store everything internally as pandas, that will make it much easier to do operations on the datasets. (And of course convert before uploading dataset / performing a run). However maybe that should be a next PR.

Reply:
I would say that this is another PR.

I would imagine that it would require quite some changes but if the consensus is to use pandas then why not. Then, the data should be only tabular data (which is probably the main use case in openml)

Reply:
Sparse data might be an issue, right?

Reply:
Sparse arrays do not have `values` attribute so we should be fine.

If you think about SparseDataFrame, I would say that we should not offer support. They are going to be [deprecated](https://github.com/pandas-dev/pandas/issues/19239) in favor of sparse dtype.

Is the liac-arff actually usable with sparse data?

Reply:
So this will not be an issue. @amueller could you elaborate on your thoughts.